spec/factories.rb

```diff
@@ -324,6 +324,10 @@
     trait :drafting_budget do
       association :group, factory: [:budget_group, :drafting_budget]
     end
+
+    trait :city_heading do
+      id 1
+    end
```

Comment:
Do you think it makes sense to, instead of forcing the ID when creating a record, change he code in the model? Maybe something like:

```ruby
def city_heading_id
  Budget::Ballot::Line.first.id
end
```
I haven't tested it and am not familiar with this part of the code, so there might a reason to force the ID I haven't heard about.

Reply:
Hi @javierm! That's a reasonable concern (hardcoding `id`s in the codebase is definitely not ideal...), however, I'm really not sure what intentions they had with that code. My understanding is that they know that in the production database the "city heading" happens to have id `1`, so this is just a hack to refer to that specific heading (at the model level, there's nothing that distinguishes a "city heading" from other headings).

In that case, the code you suggest would have the same effect, but I think it might hide the meaning of what a "city heading" is and I'm not sure if we would gain anything. Also, it would be susceptible to breaking in a non-obvious way if the default order changes (e.g. by a ` default_scope { order(created_at: :desc) }`), and might be more difficult to test.

If anything, I might change it to:
```ruby
def city_heading_id
  Budget::Ballot::Line.find_by(name: "Toda la ciudad")
end
```
as this seems to be the criterion used to identify a "city heading" in other parts of the codebase (try `grep -R 'Toda la ciudad`). I'm definitely open to any changes :), but again, there's a lot I don't know either, so you might want to check with @voodoorai2000 or @MariaCheca.

Reply:
@mlovic Interesting! Let's hope we never change the database description to "Todo el municipio" or something similar :smile:. And on my development machine the heading "Toda la ciudad" is actually the last one, with `5` as ID.

Let's wait for @voodoorai2000 or @MariaCheca to comment then. Thanks!

BTW: I made a mistake earlier. Instead of `Budget::Ballot::Line.first.id`, I meant `Budget::Heading.order(:id).first.id`. But searching by name as you suggest sounds slightly better.

Reply:
> And on my development machine the heading "Toda la ciudad" is actually the last one, with 5 as ID.

Mine too... Okay, you're definitely right. We should change that.

Also, I'm realizing that the query would need to be scoped by the ballot line's budget (different budgets can have their own "city heading").

    budget.headings.find_by(name: "Toda la ciudad")

I'll also propose creating two model methods: `Budget#city_heading` and `Budget::Heading#city_heading?` to encapsulate this ~hack~ logic, so the rest of the codebase doesn't need to know about it.

Reply:
The idea seems awesome :+1:!